pkg/plugin/framework/backup_item_action_client.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2017, 2019 the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"encoding/json"
+
+	plugin "github.com/hashicorp/go-plugin"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	api "github.com/heptio/velero/pkg/apis/velero/v1"
+	proto "github.com/heptio/velero/pkg/plugin/generated"
+	"github.com/heptio/velero/pkg/plugin/velero"
+)
+
+// BackupItemActionPlugin is an implementation of go-plugin's Plugin
+// interface with support for gRPC for the backup/ItemAction
+// interface.
+type BackupItemActionPlugin struct {
```

Comment:
this type is technically neither client nor server - it exposes the `GRPCClient` and `GRPCServer` methods. I'd probably opt to move it into its own file (and similarly for the 3 other plugin types).

Reply:
By `into its own file` do you mean name the file `backup_item_action.go` instead of `backup_item_action_client.go`? And then when I split this into `client/server` packages I'll duplicate it?

Reply:
I'd move `type BackupItemActionPlugin struct` into a file called `backup_item_action.go`, and leave `type BackupItemActionGRPCClient struct` in `backup_item_action_client.go`. And yeah, if/when we do the client/server split, then each of those packages would end up with a `backup_item_action.go` file/struct, with slightly different implementations.